src/com/trilead/ssh2/crypto/PEMDecoder.java

```diff
@@ -487,7 +489,7 @@ public static KeyPair decodeKeyPair(char[] pem, String password) throws IOExcept
 						if (password == null)
 							throw new IOException("PEM is encrypted, but no password was specified");
 
-						decryptPEM(ps, password.getBytes("ISO-8859-1"));
+						decryptPEM(ps, password.toCharArray());
```

Comment:
I see a potential issue, if the locale of the JVM is not ISO-8859-1, the password characters can be changed

Reply:
The char array is independent of the character set encoding. The native Java APIs for these use `char[]` for passwords rather than `byte[]` so that you can overwrite the array with zeros when you're done with it.